core/templates/services/platform-feature.service.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service for retriving feature flags - boolean parameters
+ * that are used to determine if features should be enabled.
+ *
+ * Once the initialization is done, the value of each feature flag is guaranteed
+ * to be constant within the page.
+ * The values are also cached in SessionStorage, so that even after page
```

Comment:
Should we use localStorage? What happens when the user opens a new tab. Will the session storage persist? As far as I know, sessionStorage is bound not only to the origin but also to the browser tab. And there are a lot of links in our app that open a new tab.

Reply:
Given that this came up in review, I suggest making this super explicit in the documentation so that others understand the context as well.

Reply:
Added descriptions at the top comments.
Also added section 7.1 in the design doc for the reason why session storage is chosen over local storage.

Reply:
Ah okay! Thanks for the explanation.